packages/teleport/src/DesktopSession/WarningDropdown.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useRef, useState } from 'react';
+import { Text, Flex, Button, Card, ButtonIcon } from 'design';
+import styled from 'styled-components';
+import { Notification } from 'shared/components/Notification';
+import { Warning, Close } from 'design/Icon';
+import { orange } from 'design/theme/palette';
+import { useClickOutside } from 'shared/hooks/useClickOutside';
+
+import type { NotificationItem } from 'shared/components/Notification';
+
+export function WarningDropdown({ warnings, onRemoveWarning }: Props) {
+  const [showDropdown, setShowDropdown] = useState(false);
+  const ref = useRef(null);
+
+  const toggleDropdown = () => {
+    if (warnings.length > 0) {
+      setShowDropdown(prevState => !prevState);
+    }
+  };
+
+  // Dropdown is always closed if there are no errors to show
+  if (warnings.length === 0 && showDropdown) setShowDropdown(false);
+
+  // Close the dropdown if it's open and the user clicks outside of it
+  const handleClickOutside = (e: MouseEvent) => {
+    if (ref.current && !ref.current.contains(e.target)) {
+      setShowDropdown(prevState => {
+        if (prevState) {
+          return false;
+        }
+      });
+    }
+  };
+  useClickOutside(ref, handleClickOutside);
+
+  return (
+    <StyledRelative ref={ref}>
+      <StyledButton
+        kind={'warning'}
+        hasWarnings={warnings.length > 0}
+        px={2}
+        onClick={toggleDropdown}
+      >
+        <Flex alignItems="center" justifyContent="space-between">
+          <StyledWarningIcon mr={2} /> {warnings.length}
+        </Flex>
+      </StyledButton>
+      {showDropdown && (
+        <StyledCard
+          mt={3}
+          p={2}
+          style={{
+            maxHeight: window.innerHeight / 4,
+          }}
+        >
+          <Flex alignItems="center" justifyContent="space-between">
+            <Text typography="h4" px={3} style={{ overflow: 'visible' }}>
+              {warnings.length} {warnings.length > 1 ? 'Warnings' : 'Warning'}
+            </Text>
+            <ButtonIcon size={1} ml={1} mr={2} onClick={toggleDropdown}>
+              <Close />
+            </ButtonIcon>
+          </Flex>
+          <StyledOverflow>
+            {warnings.map(warning => (
+              <Notification
+                my={2}
```

Comment:
Nit: if you want, you could try to replace Y margins with the flex box `gap`. 
Example from Connect [here](https://github.com/gravitational/webapps/pull/1316).
It makes margins more consistent:
`margins`
<img width="366" alt="image" src="https://user-images.githubusercontent.com/20583051/201939344-3fbe070d-9964-40fc-a0c8-2ec8f654ad52.png">
vs
`gap`
<img width="366" alt="image" src="https://user-images.githubusercontent.com/20583051/201939516-78df6c1d-fcac-407c-a12b-399ee74d0336.png">

Reply:
TIL, thanks